src/Elastic.Apm/Metrics/MetricsProvider/SystemTotalCpuProvider.cs

```diff
@@ -6,22 +6,38 @@
 using System.Linq;
 using System.Runtime.InteropServices;
 using Elastic.Apm.Api;
+using Elastic.Apm.Logging;
 
 namespace Elastic.Apm.Metrics.MetricsProvider
 {
 	internal class SystemTotalCpuProvider : IMetricsProvider, IDisposable
 	{
+		private readonly IApmLogger _logger;
 		private const string SystemCpuTotalPct = "system.cpu.total.norm.pct";
 		private readonly PerformanceCounter _processorTimePerfCounter;
 		private readonly StreamReader _procStatStreamReader;
 
-		public SystemTotalCpuProvider()
+		public SystemTotalCpuProvider(IApmLogger logger)
 		{
+			_logger = logger.Scoped(nameof(SystemTotalCpuProvider));
 			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
 			{
-				_processorTimePerfCounter = new PerformanceCounter("Processor", "% Processor Time", "_Total");
-				//The perf. counter API returns 0 the for the 1. call (probably because there is no delta in the 1. call) - so we just call it here first
-				_processorTimePerfCounter.NextValue();
+				try
+				{
+					_processorTimePerfCounter = new PerformanceCounter("Processor", "% Processor Time", "_Total");
+					//The perf. counter API returns 0 the for the 1. call (probably because there is no delta in the 1. call) - so we just call it here first
+					_processorTimePerfCounter.NextValue();
+				}
+				catch (Exception e)
+				{
+					_logger.Error()?.LogException(e, "Failed instantiating PerformanceCounter "
+						+ "- please make sure the current user has permissions to read performance counters. E.g. make sure the current user is member of "
+						+ "the 'Performance Monitor Users' group");
+				}
+				finally
+				{
+					_processorTimePerfCounter?.Dispose();
```

Comment:
It seems that `_processorTimePerfCounter = null;` is missing, right?

Reply:
Added. I think it remans `null`, so this was more like a "just in case let's dispose it", but you are right: if we can dispose it, we should also set it to `null`. 👍